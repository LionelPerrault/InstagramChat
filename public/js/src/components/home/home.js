import React from 'react'
import { FadeIn } from 'animate-components'
import Title from '../others/title'
import { getUnreadNotifications } from '../../store/actions/notification-a'
import { connect } from 'react-redux'
import $ from 'jquery'
import { getFeed } from '../../store/actions/post-a'
import Loading from '../others/loading'
import Nothing from '../others/nothing'
import Post from '../post/post'
import End from '../others/end'
import MessageNotify from '../others/message-notify'
import Suggested from '../others/suggested/suggested'
import CreateGroup from '../group/create-group/create-group'
import PostItTeaser from '../post/post-it/post-it-teaser'
import { getUnreadMessages } from '../../store/actions/message-a'

@connect(store => {
  return {
    feed: store.Post.feed,
    posted: store.Post.posted
  }
})

export default class Home extends React.Component {

  state = {
    loading: true,
  }

  componentDidMount = () => {
    let { dispatch } = this.props
    dispatch(getFeed())
    dispatch(getUnreadNotifications())
    dispatch(getUnreadMessages())
  }

  componentWillReceiveProps = () =>
    this.setState({ loading: false })

  render() {
    let
      username = $('.data').data('username'),
      { loading } = this.state,
      { feed, posted } = this.props,
      len = feed.length,
      map_feed = feed.map(f =>
        <Post key={f.post_id} {...f} when='feed' />
      )

    return (
      <div>

        <Title value='Home' />

        <FadeIn duration='300ms'>

          { loading ? <Loading/> : null }

          <div className={`senapati home_senapati ${loading ? 'cLoading' : ''}`} >

            <div className='prajkumar'>
              <PostItTeaser type='user' />

              <div
                className='posts_div'
                style={{ marginTop: len == 0 ? 10 : 0 }}
              >

                {
                  posted ?
                    <MessageNotify
                      url={`/profile/${username}`}
                      btnText='Checkout'
                      message='Checkout your profile page to see your recent post!!'
                    />
                    : null
                }

                {
                  len == 0 ?
                    <Nothing mssg="Looks like you're new, Follow some to fill up your feed or post from above options!!" />
                    : <FadeIn duration='500ms'>{ map_feed }</FadeIn>
                }
              </div>

              { len != 0 ? <End/> : null }

            </div>

            <div className='srajkumar'>
              <Suggested when='home' />
              <CreateGroup/>
            </div>

          </div>

        </FadeIn>

      </div>
    )
  }
}
