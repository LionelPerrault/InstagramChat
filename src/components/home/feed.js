import React, { Fragment } from 'react'
import { connect } from 'react-redux'
import End from '../others/end'
import Post from '../post/post/post'
import MapPosts from '../post/map-posts/map-posts'

const Feed = ({ feed }) => {
  let len = feed.length
  let map_feed = feed.map(f => <Post key={f.post_id} {...f} when="feed" />)

  let nothingMssg =
    "Looks like you're new, Follow some to fill up your feed or post from above options!!"

  return (
    <Fragment>
      <div className="posts_div" style={{ marginTop: len == 0 ? 10 : 0 }}>
        <MapPosts posts={map_feed} nothingMssg={nothingMssg} />
      </div>

      {len != 0 && <End />}
    </Fragment>
  )
}

const mapStateToProps = state => ({
  feed: state.Post.feed,
})

export default connect(mapStateToProps)(Feed)
