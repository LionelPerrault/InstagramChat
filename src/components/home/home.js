import React, { Component } from 'react'
import { FadeIn } from 'animate-components'
import Title from '../others/title'
import { getUnreadNotifications } from '../../actions/notification'
import { connect } from 'react-redux'
import { getFeed } from '../../actions/post'
import Suggested from '../others/suggested/suggested'
import CreateGroup from '../group/create-group/create-group'
import PostItTeaser from '../post/post-it/post-it-teaser'
import { getUnreadMessages } from '../../actions/message'
import PopularHashtags from '../hashtag/popular-hashtags'
import { Instagram } from 'react-content-loader'
import Feed from './feed'

class Home extends Component {
  state = {
    loading: true,
  }

  componentDidMount = () => {
    let { dispatch } = this.props
    dispatch(getFeed())
    dispatch(getUnreadNotifications())
    dispatch(getUnreadMessages())
  }

  componentWillReceiveProps = () => this.setState({ loading: false })

  render() {
    let { loading } = this.state

    return (
      <div>
        <Title value="Home" />

        <FadeIn duration="300ms">
          <div className="senapati home_senapati">
            <div className="prajkumar">
              <PostItTeaser type="user" disabled={loading} />

              {loading && (
                <div style={{ marginTop: 20 }}>
                  <Instagram />
                  <Instagram />
                  <Instagram />
                </div>
              )}

              <Feed />
            </div>

            <div className="srajkumar">
              <Suggested when="home" />
              <PopularHashtags />
              {!loading && <CreateGroup />}
            </div>
          </div>
        </FadeIn>
      </div>
    )
  }
}

const mapStateToProps = store => ({
  store,
})

export default connect(mapStateToProps)(Home)
export { Home as PureHome }
