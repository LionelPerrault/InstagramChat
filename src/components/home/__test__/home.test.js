import React from 'react'
import { PureHome } from '../home'
import ShallowRenderer from 'react-test-renderer/shallow'
import { shallow } from 'enzyme'

describe('Home Component', () => {

  // shallow snapshot
  it('should match snapshot', () => {
    const renderer = new ShallowRenderer()
    const tree = renderer.render(
      <PureHome/>
    )
    expect(tree).toMatchSnapshot()
  })

  it('should show <Instagram/> loader when loading=true', () => {
    const wrapper = shallow(
      <PureHome
        dispatch={jest.fn()}
      />
    )
    wrapper.setState({ loading: true })
    expect(wrapper.find('InstagramStyle').length).toBe(3)
  })

})
