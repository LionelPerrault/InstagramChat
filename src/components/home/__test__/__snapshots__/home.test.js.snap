// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component should match snapshot 1`] = `
<div>
  <Connect(Title)
    value="Home"
  />
  <FadeIn
    as="div"
    backfaceVisible="visible"
    delay="0s"
    direction="normal"
    duration="300ms"
    fillMode="none"
    iterations="1"
    playState="running"
    timingFunction="ease"
  >
    <div
      className="senapati home_senapati"
    >
      <div
        className="prajkumar"
      >
        <Connect(PostItTeaser)
          disabled={true}
          type="user"
        />
        <div
          style={
            Object {
              "marginTop": 20,
            }
          }
        >
          <InstagramStyle />
          <InstagramStyle />
          <InstagramStyle />
        </div>
        <Connect(Feed) />
      </div>
      <div
        className="srajkumar"
      >
        <Connect(Suggested)
          when="home"
        />
        <Connect(PopularHashtags) />
      </div>
    </div>
  </FadeIn>
</div>
`;
